pkg/server/server.go

```diff
@@ -128,18 +148,15 @@ func (s *Server) buildVcsToArgoMap() error {
 
 	ctx := context.TODO()
 
-	result := pkg.NewVcsToArgoMap()
-
 	argoClient := argo_client.GetArgoClient()
 
 	apps, err := argoClient.GetApplications(ctx)
 	if err != nil {
 		return errors.Wrap(err, "failed to list applications")
 	}
 	for _, app := range apps.Items {
-		result.AddApp(app)
+		s.cfg.VcsToArgoMap.AddApp(&app)
```

Comment:
Should we be worried about memory constraints as number of apps increase? This is not an immediate problem tho, so it's fine.